app/code/Magento/Indexer/Test/Mftf/ActionGroup/IndexerActionGroup/AdminIndexerSetUpdateOnSaveActionGroup.xml

```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+<actionGroups xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/actionGroupSchema.xsd">
+    <actionGroup name="AdminIndexerSetUpdateOnSave">
+        <annotations>
+            <description>Goes to the Index Management page. Checks the provided Indexer Name. Selects 'Update on Save'. Clicks on Submit.</description>
+        </annotations>
+        <arguments>
+            <argument name="indexerName" type="string"/>
+        </arguments>
+
+        <amOnPage url="{{_ENV.MAGENTO_BACKEND_NAME}}/indexer/indexer/list/" stepKey="amOnIndexManagementPage2"/>
+        <waitForPageLoad stepKey="waitForIndexManagementPageToLoad2"/>
+        <click selector="{{AdminIndexManagementSection.indexerCheckbox(indexerName)}}" stepKey="selectIndexer2"/>
+        <selectOption selector="{{AdminIndexManagementSection.massActionSelect}}" userInput="change_mode_onthefly" stepKey="selectUpdateOnSave"/>
+        <click selector="{{AdminIndexManagementSection.massActionSubmit}}" stepKey="submitIndexerForm2"/>
+        <!-- No re-indexing is done as part of this actionGroup since the test required no re-indexing -->
+        <waitForPageLoad stepKey="waitForSave2"/>
```

Comment:
```suggestion
        <magentoCli command="indexer:set" arguments="{{indexerCode}}" stepKey="..."/>
        <magentoCli command="cache:clean" arguments"config" stepKey="..."/>
```

Reply:
Hello @lbajsarowicz ...
I don't think that we need to replace the functional test for indexer with Magento cli cause we are not able to test properly that indexer admin grid is working properly or not. Let me know your thoughts
Thanks

Reply:
Yes, here you are right.